content/en/platform/corda/5.2/deploying-operating/keys/_index.md

```diff
@@ -0,0 +1,196 @@
+---
+description: "Learn how to upload and rotate Corda 5 wrapping keys."
+title: "Managing Wrapping Keys"
+date: '2024-02-12'
+menu:
+  corda52:
+    identifier: corda52-cluster-wrapping-keys
+    parent: corda52-cluster
+    weight: 3025
+---
+
+# Managing Wrapping Keys
+
+As described in the [Architecture for Cluster Administrators]({{< relref "../../key-concepts/cluster-admin/_index.md#key-management" >}}) section, Corda keys are stored in the cluster and virtual node `Crypto` databases and encrypted at rest with wrapping keys.
+These managed wrapping keys are in turn wrapped by a master wrapping key.
+This section describes how Corda stores the master wrapping key and how to rotate master and managed wrapping keys. It contains the following:
+
+{{< toc >}}
+
+## Configuring the Master Wrapping Key
+
+The master wrapping key, or the information required to generate this key, must be stored and managed outside Corda.
+You can achieve this in one of the following ways:
+
+* Pass a passphrase and salt, to generate the master key, into the crypto worker processes. For more information, see [Default Secrets Service]({{< relref "../../deploying-operating/deployment/deploying/_index.md#default-secrets-service" >}}).
```

Comment:
`crypto worker processes` is an implementation detail customers don't really need to understand at this level. Also if you you use this method you aren't really storing outside Corda as the previous line suggests.

Suggestion:
The master wrapping key must be stored and managed outside Corda when running a production deployment. For this an external key management system is used to retrieve the master key when required. For more information, see [External Secrets Service]({{< relref "../../deploying-operating/deployment/deploying/_index.md#external-secrets-service" >}}).

For testing, development or other non production uses, a passphrase and salt can be provided to generate the master key, and the master key will then be stored by Corda. For more information, see [Default Secrets Service]({{< relref "../../deploying-operating/deployment/deploying/_index.md#default-secrets-service" >}}).